semgrep-core/tests/OTHER/rules/const_prop_toplevel.py

```diff
@@ -0,0 +1,10 @@
+# ruleid: const_prop_toplevel
+x = "ls"
+
+foo = "ls"
+# ruleid: const_prop_toplevel
+x = foo
+
+foo = "ls"
+# ruleid: const_prop_toplevel
+x = foo
```

Comment:
Maybe we need to reconsider this, because this test does not justify all the code that is being changed here. In `Match_tainting_mode` we have this hack:
```
        let xs = AST_to_IL.stmt lang (G.stmt1 ast) in
        let flow = CFG_build.cfg_of_stmts xs in
        Dataflow_tainting.fixpoint ~fun_env taint_config flow |> ignore)
```
And I wonder if a similar hack could be used for constant propagation to achieve something very similar to what this PR currently achieves? Or am I wrong here?